### PR TITLE
Doc updates

### PR DESCRIPTION
**Pull request by bootstraponline**
- Add note about GREYCondition
- Explain how to setup Ruby

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -366,7 +366,7 @@ EarlGrey.select(elementWithMatcher:matcher)
 
 The best way is to [setup synchronization](features.md#synchronization) so that EarlGrey automatically waits
 for elements to appear. As a work around for when that's not possible, `GREYCondition` and `waitWithTimeout`
-are available. The following is an example of waiting for a collection view to populate.
+are available. The following is an example of waiting for a collection view to populate. Note that `pollInterval` should be > 0 to avoid excessive hierarchy scans slowing down the main thread.
 
 ```swift
```
Comment (khandpur) [inline docs/faq.md]: if it's not too much to ask could you all an objective-c way of doing it?

```
GREYCondition *waitCondition = [GREYCondition conditionWithName:@"Wait for UICollectionView to populate" block:^BOOL {
  NSError *error;
  // Checking if collection view exists in the UI hierarchy.
  [[EarlGrey selectElementWithMatcher:collectionViewMatcher]  
      assertWithMatcher:grey_notNil() error:&error];
  return error == nil;
}];

// Wait until 5 seconds for the view.
BOOL populated = [waitCondition waitWithTimeout:5.0 pollInterval:1.0];
GREYAssertTrue(populated, @"Failed to populate UICollectionView in 5 seconds");
```

File: docs/install-and-run.md [hunk 2]
```diff
@@ -285,3 +285,40 @@ compatible with EarlGrey.
 can run it by just clicking on the **run** icon or by Ctrl+clicking the test name and then selecting
 **Test "testPresenceOfKeyWindow"**. Because this is a regular unit test, you can place breakpoints in
 your test and in your application code and use the built-in tools seamlessly.
+
+### Ruby Installation
```
Comment (khandpur) [inline docs/install-and-run.md]: Thanks for adding this. You should link to to one of the install steps above so user can jump to this section before they install EarlGrey gem, etc.

File: docs/install-and-run.md [hunk 3]
```diff
@@ -285,3 +285,40 @@ compatible with EarlGrey.
...
+
+If you already have a working Ruby install, skip this part.
```
Comment (tirodkar) [inline docs/install-and-run.md]: Could you please number each step and then indent 3 spaces for every step?

For eg.

### Ruby Installation

If you already have a working Ruby install, skip this part.

   1. Install Ruby Env.
      ```
          brew update
          brew install rbenv
      ```

**Comment by bootstraponline** [comment]
Updated with:

* Section link
* Objective-C example
* Numbered & indented steps

**Pull request merged**